### PR TITLE
fix(#399): scope featured-example IDs per-mount + strip harness from Show-code snippet

### LINKED ISSUE #399: Bug: Duplicate element IDs across Overview and Examples mounts break ARIA on tag-input and textarea
## Summary

The playground doc page mounts a component's "featured" example twice: once in the Overview live-preview container and once in the Examples section. Both mounts use the same hardcoded `id` prop values from the example source file, so two elements with identical IDs exist simultaneously in the DOM. Duplicate IDs are invalid HTML and break `aria-labelledby` / `aria-describedby`.

## Findings

| # | Component | Where | What is wrong |
|---|---|---|---|
| 1 | `tag-input` | Overview live preview vs. "Basic tag input" example | Both mounts render `id="basic-tag-input"`, `id="basic-tag-input-label"`, `id="basic-tag-input-tags"`, and `id="basic-tag-input-description"`. The second mount's ARIA associations are broken. |
| 2 | `textarea` | Overview live preview vs. "Basic textarea" example | Both mounts render `id="bio"`. The "Bio" label links to the first textarea only. |

A related issue: the `subscription-badge` page has two "Show code" accordion triggers both rendered with `id="source-header"`, breaking `aria-controls` and `aria-labelledby`. This is a doc-page template defect affecting any component with multiple examples.

## Implementation plan

1. **Playground scaffold** — `packages/playground/src/component-page.svelte`: Pass a unique mount-scoped ID prefix into each scenario mount (the `mountScenario` attachment already knows the container DOM id: `overview-mount-<scenario>` vs `example-mount-<scenario>`). Expose this as a prop or slot context so example components can suffix their IDs.

2. **Example source files** — Update the two affected examples:
   - `packages/playground/src/examples/tag-input/basic.example.svelte`: use the dynamically scoped ID prefix.
   - `packages/playground/src/examples/textarea/basic.example.svelte`: use the dynamically scoped ID prefix.

3. **Accordion trigger IDs** — In `component-page.svelte`, locate the "Show code" accordion trigger and change the hard-coded `id="source-header"` to a dynamic value incorporating the example index or slug, e.g. `id="source-header-{scenario.id}"`. This prevents the duplicate-ID bug on any component page with multiple examples.

4. After fixing, run an axe pass on both doc pages (tag-input and textarea) and confirm `duplicate-id-aria` no longer fires.

5. Add a Playwright test or axe-based assertion that fails if duplicate IDs appear in the rendered page DOM for any component with an Overview live preview.

---
_Filed from an automated visual + interactive review (light + dark, keyboard, hover, open/close) of these component pages, driven through Chrome against a static export of the playground. Findings in this batch were adversarially re-verified in a second browser pass before filing._

### PR DESCRIPTION
**Pull request by stevekinney**
## Summary

The component doc page (`/page/<slug>`) mounts a component's featured example **twice** in non-snapshot mode — once in the Overview live-preview container (`#overview-mount-<scenario>`) and once in the Examples section (`#example-mount-<scenario>`). The affected example files hardcoded static element `id` attributes (`id="bio"`, `id="basic-tag-input"`, …), so both mounts emitted **identical element IDs** into the same document. Duplicate IDs are invalid HTML and break `aria-labelledby` / `aria-describedby` / `<label for>` (axe `duplicate-id-aria`).

Closes #399.

## The fix

Each mount now receives a unique `mountIdPrefix` prop (its container's DOM id) from `component-page.svelte`. Every affected example derives the element ids it emits from that prefix, so the two mounts produce fully distinct id trees. The canonical idiom (the only form):

```svelte
let { mountIdPrefix }: { mountIdPrefix?: string } = $props();
const uid = $props.id();
let fieldId = $derived(`${mountIdPrefix ?? uid}-field`);
```

- `$props.id()` is Svelte 5's per-instance stable id — so even two standalone copies of an example (no injected prefix) never collide. It can't be a destructure default (placement rule), hence the separate `const uid` + `mountIdPrefix ?? uid`.
- 28 featured examples converted; most cinder field components derive their `-label` / `-description` / `-error` / `-listbox` child ids from the single `id`, so prefixing that one id cascades.

### Keeping the "Show code" snippet clean

"Show code" serves the **raw** `.example.svelte` file (`/example-src/:name/:scenario`). The mount-isolation harness (`mountIdPrefix` / `$props.id()` / the derived-id binding) is doc-page plumbing a reader copying the snippet should never see. A new `stripExampleHarness` removes the harness at **serve time** and rewrites each derived-id binding site to the static literal it would resolve to (`id={fieldId}` → `id="field"`), so the copyable snippet is clean, idiomatic consumer code.

The strip is **fail-closed**: if any removed identifier still appears in a `{…}` expression, or any harness marker (`mountIdPrefix` / `$props.id()`) survives, it throws — and the `/example-src/` route turns that throw into a diagnostic 500 (`exampleSnippetResponse`) rather than serving half-stripped code.

## Why source-level tests, not a browser test

The shared `componentPage.open()` fixture forces `?snapshot=1`, and in snapshot mode **only** the `example-mount-*` container renders — the Overview mount is suppressed. So the dup-id bug cannot reproduce through the standard fixture, and a snapshot-mode Playwright test would pass even on the OLD buggy code. The bug and fix are deterministic source-level properties, so:

- `featured-example-mount-id.test.ts` — pins the per-mount-id contract on all 28 examples (declares `mountIdPrefix`, falls back to `$props.id()`, derives ≥1 id from the prefix, no longer hardcodes the old literal) + a `derivedId(prefix, suffix)` distinctness model.
- `strip-example-harness.test.ts` — fixtures per harness shape, route-glue 200/500 integration tests, and a sweep that strips **every** real harnessed example and compiles the output with `svelte/compiler` to prove it stays valid copy-pasteable Svelte. Sweep count is cross-checked against the featured registry so a moved/renamed dir can't silently zero it.

`markdown-editor-toolbar.playwright.ts` updated: the editor locator moved from the now-per-mount-namespaced `#playground-markdown-editor` to the stable `.markdown-editor-wrapper` component-root class.

## Review

Reviewed by the committee (svelte-expert, typescript-expert, testing-expert, simplicity-engineer, ux-designer, junior-engineer, + Codex/gpt-5.4). Round-1 must-fixes (route fail-closed glue + test, eliminate the hand-curated bare-id side table, break→throw on malformed regex, residual-marker scan, replacement callbacks, `$`-prefix identifier boundaries) all implemented; round-2 re-review reached **unanimous APPROVED**.

## Verification

- `strip-example-harness.test.ts` + `featured-example-mount-id.test.ts`: 138 pass, 0 fail.
- Full pre-commit + pre-push suites (typecheck + tests for `@lostgradient/cinder`, `@cinder/playground`, `@cinder/testing`): green.
- `components:check`: OK (zero manifest drift).

---

> [!NOTE]
> **Medium Risk**
> Touches published example manifest `code` and the Show-code HTTP path; mistakes could ship broken snippets, but fail-closed stripping and broad tests reduce that risk.
> 
> **Overview**
> Fixes **duplicate element IDs** when the component doc page mounts the same featured example in Overview and Examples (#399). **28 playground examples** now accept `mountIdPrefix` (from `component-page.svelte`) and derive DOM `id`/`name`/`target` from `${mountIdPrefix ?? $props.id()}-…` instead of hardcoded literals.
> 
> Adds **`stripExampleHarness`** so doc-only harness code never appears in copy-paste surfaces: the example generator’s shipped `code` field and the playground **`/example-src/`** route both strip `mountIdPrefix` / `$props.id()` plumbing and rewrite bindings to static literals (e.g. `id={fieldId}` → `id="field"`). Unrecognized harness shapes **fail closed** (extraction error or HTTP 500), blocking half-stripped snippets.
> 
> **Regenerated** `*.examples.json` artifacts reflect the cleaned snippets. **Tests** cover the transform, extractor integration, featured-example source contracts, route glue, and a full harnessed-example compile sweep; **markdown-editor** Playwright locators use `.markdown-editor-wrapper` instead of a per-mount id.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c7dfb16904f5363d1c096acff1cc7cab6c351a67. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/components/src/components/accordion/accordion.examples.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
...
+      "code": "<script lang=\"ts\">\n  import { Accordion } from '@lostgradient/cinder/accordion';\n\n  let { mountIdPrefix }: { mountIdPrefix?: string } = $props();\n  const uid = $props.id();\n  let item1Id = $derived(`${mountIdPrefix ?? uid}-item-1`);\n  let item2Id = $derived(`${mountIdPrefix ?? uid}-item-2`);\n  let item3Id = $derived(`${mountIdPrefix ?? uid}-item-3`);\n\n  let expandedIds = $state<string[]>([]);\n</script>\n\n<Accordion bind:expandedIds>\n  <Accordion.Item id={item1Id} title=\"What is Cinder?\">\n    Cinder is a Svelte 5 design system built for composability and accessibility.\n  </Accordion.Item>\n  <Accordion.Item id={item2Id} title=\"How do I install it?\">\n    Run <code>bun add @lostgradient/cinder</code> and import components from\n    <code>@lostgradient/cinder</code> or via subpath imports.\n  </Accordion.Item>\n  <Accordion.Item id={item3Id} title=\"Does it support SSR?\">\n    Yes — every component is SSR-safe and renders meaningful HTML before hydration.\n  </Accordion.Item>\n</Accordion>\n"
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/accordion/accordion.examples.json]: Fixed in c7dfb169. The strip now runs at **manifest generation** (`extractExampleFile` in `packages/components/scripts/generate-component-examples.ts`), not only at the playground serve route, so the shipped `@lostgradient/cinder/<component>/examples` `code` fields are clean, copy-pasteable Svelte with no `mountIdPrefix`/`$props.id()` plumbing. `stripExampleHarness` moved to `packages/components/scripts/lib/` as the single source of truth (the generator owns the published artifact; the playground server reaches back via the established cross-package relative import). All 28 manifests + `components.json` regenerated — `components:check` reports zero drift and zero harness leakage. The strip is fail-closed: an unrecognized binding shape now surfaces as a hard extraction error that blocks `components:generate` (and publishing) rather than shipping half-stripped code.

File: packages/playground/src/strip-example-harness.test.ts [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+import { describe, expect, it } from 'bun:test';
+import { readdir } from 'node:fs/promises';
+import { basename, dirname, join } from 'node:path';
+
+import { compile } from 'svelte/compiler';
+
+import { FEATURED_EXAMPLES } from './examples/featured-example-mount-id.test.ts';
+import { exampleSnippetResponse } from './playground-server.ts';
+import { stripExampleHarness } from './strip-example-harness.ts';
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/playground/src/strip-example-harness.test.ts]: Fixed in c7dfb169. `FEATURED_EXAMPLES` and the source-shape predicates moved into a plain (non-test) module, `packages/playground/src/examples/featured-examples.ts`. Both the `featured-example-mount-id.test.ts` guard and the strip-harness full-corpus sweep now import the registry from there, so no test file imports another test file's exports.
